share/eds/cache/accessor_cache.go

```diff
@@ -55,6 +64,47 @@ func (s *accessorWithBlockstore) Reader() io.Reader {
 	return s.shardAccessor.Reader()
 }
 
+func (s *accessorWithBlockstore) addRef() error {
+	s.RLock()
+	defer s.RUnlock()
+	if s.isClosed {
+		// item is already closed and soon will be removed after all refs are released
+		return errCacheMiss
+	}
+	if s.refs.Add(1) == 1 {
+		// there were no refs previously and done channel was closed, reopen it by recreating
+		s.done = make(chan struct{})
+	}
+	return nil
+}
+
+func (s *accessorWithBlockstore) removeRef() {
+	s.RLock()
+	defer s.RUnlock()
+	if s.refs.Add(-1) <= 0 {
+		close(s.done)
+	}
+}
+
+func (s *accessorWithBlockstore) close() error {
+	s.Lock()
+	s.isClosed = true
+	done := s.done
+	s.Unlock()
```

Comment:
Because it could case race condition, when `close()` will resolve by closed done channel, which could allow closure of underlying accessore. While in parallel new ref could be created by `addRef`, making underlying accessor being in use.
So  `isClosed` is needed to prevent new ref, after accessor closure has been initialised.